### PR TITLE
Add embedder unit test that reproduces https://github.com/dart-lang/sdk/issues/46275

### PR DESCRIPTION
**Pull request by a-siva**
Add unit test case to reproduce the shutdown crash demonstrated in

https://github.com/dart-lang/sdk/issues/46275

### CODE REVIEW

**Comment by a-siva** [comment]
This PR should land only after https://dart-review.googlesource.com/c/sdk/+/206800 lands in the Dart repo and that change has been rolled into the engine.

File: shell/platform/embedder/fixtures/main.dart [hunk 1]
```diff
@@ -9,6 +9,7 @@ import 'dart:isolate';
 import 'dart:ffi';
 import 'dart:core';
 import 'dart:convert';
+import 'dart:io';
```
Comment (zanderso) [inline shell/platform/embedder/fixtures/main.dart:12]: small cleanup: sort imports.

note to self: we need to figure out how to run the analyzer/linter over these fixtures.

**Review by zanderso — COMMENTED** [review]
change lgtm, but we can rebase after the fix from the Dart VM rolls in to get the presubmits to be green.

**Review by zanderso — APPROVED** [review]

File: shell/platform/embedder/fixtures/main.dart (hunk 1 above)
Comment (a-siva) [reply shell/platform/embedder/fixtures/main.dart:9]: Done.

**Pull request merged**